jruby1/plan.sh

```diff
@@ -31,7 +31,7 @@ do_build() {
 
 do_install() {
   cp -R ./* "$pkg_prefix/"
-  fix_interpreter "$pkg_prefix/bin/*" core/coreutils bin/env
+  fix_interpreter "$pkg_prefix/bin/*" core/bash bash
```

Comment:
This works for bash, but I think many of the rest of these may break now?

```
[29][default:/src:1]# file /hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/*
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/ast:         a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/gem:         a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/install_doc: a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/irb:         a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jgem:        a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jirb:        a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jirb_swing:  a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jruby:       a /hab/pkgs/core/bash/4.3.42/20170513213519/bin/bash script, ASCII textexecutable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jruby.bash:  a /hab/pkgs/core/bash/4.3.42/20170513213519/bin/bash script, ASCII textexecutable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jruby.bat:   DOS batch file, ASCII text, with CRLF line terminators
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jruby.dll:   PE32 executable (DLL) (console) Intel 80386 (stripped to external PDB),for MS Windows
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jruby.exe:   PE32 executable (console) Intel 80386 (stripped to external PDB), for MS Windows
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jruby.sh:    POSIX shell script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jrubyc:      a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jrubyd.bat:  DOS batch file, ASCII text, with CRLF line terminators
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/jrubyw.exe:  PE32 executable (GUI) Intel 80386 (stripped to external PDB), for MS Windows
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/rake:        a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/rdoc:        a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/ri:          a /usr/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/ruby:        symbolic link to jruby
/hab/pkgs/core/jruby1/1.7.27/20171206232030/bin/testrb:      a /usr/bin/env jruby script, ASCII text executable
```

Reply:
Or rather "may still be broken"

Reply:
still be broken is more likely 😄 

if this is a blocker for merging I'll fix it, otherwise I'd prefer to let that cleanup happen in a future round because this issue is affecting some critical-path work right now

Reply:
I think this also fixes the problem:

```
diff --git a/jruby1/plan.sh b/jruby1/plan.sh
index c09ee41..d9ede2c 100644
--- a/jruby1/plan.sh
+++ b/jruby1/plan.sh
@@ -31,7 +31,7 @@ do_build() {

 do_install() {
   cp -R ./* "$pkg_prefix/"
-  fix_interpreter "$pkg_prefix/bin/*" core/bash bash
+  fix_interpreter "$pkg_prefix/bin/*" core/coreutils env

   # Remove *.so for other platforms...they cause `do_strip()' to fail
   # with `Unable to recognise the format' errors
```

At least, if the command is executed via `hab pkg exec`

Reply:
```
[43][default:/src:127]# file /hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/*
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/ast:         a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/gem:         a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/install_doc: a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/irb:         a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jgem:        a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jirb:        a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jirb_swing:  a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jruby:       a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env bash script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jruby.bash:  a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env bash script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jruby.bat:   DOS batch file, ASCII text, with CRLF line terminators
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jruby.dll:   PE32 executable (DLL) (console) Intel 80386 (stripped to external PDB), for MS Windows
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jruby.exe:   PE32 executable (console) Intel 80386 (stripped to external PDB), for MS Windows
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jruby.sh:    POSIX shell script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jrubyc:      a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jrubyd.bat:  DOS batch file, ASCII text, with CRLF line terminators
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/jrubyw.exe:  PE32 executable (GUI) Intel 80386 (stripped to external PDB), for MS Windows
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/rake:        a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/rdoc:        a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/ri:          a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/ruby:        symbolic link to jruby
/hab/pkgs/core/jruby1/1.7.27/20171207000111/bin/testrb:      a /hab/pkgs/core/coreutils/8.25/20170513213226/bin/env jruby script, ASCII text executable
```

Reply:
the guidance I got from @robbkidd and @elliott-davis is that we want to stop using `env` where possible and go straight to `bash`

Reply:
> go straight to `bash`

... or whatever the script is using `env` to locate.

So maybe the plan could use more `fix_interpreter` even if it's old-school-sed style (because `fix_interpreter` won't accept the current package being built as input):

```
 do_install() {
   cp -R ./* "$pkg_prefix/"
-  fix_interpreter "$pkg_prefix/bin/*" core/coreutils bin/env
+  fix_interpreter "$pkg_prefix/bin/*" core/bash bash
+  for binstub in ${pkg_prefix}/bin/*; do
+    [[ -f $binstub ]] && sed -e "s#/usr/bin/env jruby#${pkg_prefix}/bin/jruby#" -i "$binstub"
+  done
```

... results in ...

```
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/ast:         a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/gem:         a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/install_doc: a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/irb:         a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jgem:        a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jirb:        a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jirb_swing:  a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby:       a /hab/pkgs/core/bash/4.3.42/20170513213519/bin/bash script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby.bash:  a /hab/pkgs/core/bash/4.3.42/20170513213519/bin/bash script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby.bat:   DOS batch file, ASCII text, with CRLF line terminators
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby.dll:   PE32 executable (DLL) (console) Intel 80386 (stripped to external PDB), for MS Windows
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby.exe:   PE32 executable (console) Intel 80386 (stripped to external PDB), for MS Windows
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby.sh:    POSIX shell script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jrubyc:      a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jrubyd.bat:  DOS batch file, ASCII text, with CRLF line terminators
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jrubyw.exe:  PE32 executable (GUI) Intel 80386 (stripped to external PDB), for MS Windows
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/rake:        a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/rdoc:        a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/ri:          a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/ruby:        a /hab/pkgs/core/bash/4.3.42/20170513213519/bin/bash script, ASCII text executable
/hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/testrb:      a /hab/pkgs/robbkidd/jruby1/1.7.27/20171207025047/bin/jruby script, ASCII text executable
```